jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -159,6 +159,18 @@ default void clear()
         BufferUtil.clear(getByteBuffer());
     }
 
+    /**
+     * {@link #release() Release} the buffer in a way that will remove it from any pool that it may be in.
+     * If the buffer is not in a pool, calling this method is equivalent to calling {@link #release()}.
+     * Calling this method satisfies any contract that requires a call to {@link #release()}.
+     * @return {@code true} if a call to {@link #release()} would have returned {@code true}.
+     * @see #release()
+     */
+    default boolean remove()
```

Comment:
I very much don't like this new RBB method, for at least these reasons:

First, do we want to introduce this API change in 12.0 but not have it in 12.1? That doesn't sound right to me.

Then I don't think conflating the removal from the pool with the ref counter tracking is the right way to go: when `failed/abort` is called between `write` and `succeeded` the RBB should be removed from the pool but its ref counter should be left untouched as we're still waiting for the `succeeded` call. But there's also the complicating factor of the `write` / `failed` sequence that should release but not remove.

Reply:
@lorban The problem without having this method is two fold:
 + Buffers cannot be wrapped (as in tacking apps) as that will prevent downcasting to find a removable buffer.   Perhaps a standard wrapper might help, but ultimately a buffer could be wrapped strangely.
 + We'd have to pass the pool around in places, so that we have something to remove the buffer from

Reply:
I will remove the API for 12.0 and e unwrapping.   We can review in 12.1